docs/src/user_docs/assembly/cryptographic_operations.md

```diff
@@ -10,4 +10,4 @@ Miden assembly provides a set of instructions for performing common cryptographi
 | rphash  <br> - *(16 cycles)*        | [B, A, ...]     | [C, ...]       | $C \leftarrow hash(A,B)$ <br> where, $hash()$ computes a 2-to-1 Rescue Prime Optimized hash. |
 | mtree_get  <br> - *(9 cycles)*     | [d, i, R, ...]  | [V, R, ...] |  Verifies that a Merkle tree with root $R$ opens to node $V$ at depth $d$ and index $i$. Merkle tree with root $R$ must be present in the advice provider, otherwise execution fails. |
 | mtree_set <br> - *(29 cycles)*      | [d, i, R, V', ...] | [V, R', ...] | Updates a node in the Merkle tree with root $R$ at depth $d$ and index $i$ to value $V'$. $R'$ is the Merkle root of the resulting tree and $V$ is old value of the node. Merkle tree with root $R$ must be present in the advice provider, otherwise execution fails. At the end of the operation Merkle tree in the advice provider with root $R$ is replaced with the Merkle tree with root $R'$. |
-| mtree_cwm <br> - *(12 cycles)*      | [d, i, R, V, ...] | [R', V, R, ...] | Copies a Merkle tree with root $R$ and updates a node at depth $d$ and index $i$ in the copied tree to value $V$. $R'$ is the Merkle root of the new tree. Merkle tree with root $R$ must be present in the advice provider, otherwise execution fails. At the end of the operation the advice provider will contain both Merkle trees. |
+| mtree_cwm <br> - *(29 cycles)*      | [d, i, R, V', ...] | [V, R', ...] | Copies a Merkle tree with root $R$ and updates a node at depth $d$ and index $i$ in the copied tree to value $V'$. $R'$ is the Merkle root of the new tree and $V$ is old value of the node. Merkle tree with root $R$ must be present in the advice provider, otherwise execution fails. At the end of the operation the advice provider will contain both Merkle trees. |
```

Comment:
Note: The format of the instruction `mtree_cwm` is now the same as `mtree_set`. The rationale is that all the inputs provided by the user can be easily dup'ed if necessary, and the instruction should consume the inputs.

Let me know if that should be changed.

Reply:
I agree with the rationale

Reply:
@bobbinth wdyt?

Reply:
I think this is fine. It is likely that this will be slightly less efficient than leaving the old root on the stack - but I think we can give up a bit of efficiency for the sake of consistency. If it becomes a problem, we can revisit it later.